sw/device/tests/crypto/cryptotest/firmware/hash.c

```diff
@@ -0,0 +1,150 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "sw/device/lib/crypto/include/hash.h"
+
+#include "sw/device/lib/base/memory.h"
+#include "sw/device/lib/base/status.h"
+#include "sw/device/lib/crypto/impl/integrity.h"
+#include "sw/device/lib/crypto/impl/keyblob.h"
+#include "sw/device/lib/crypto/include/datatypes.h"
+#include "sw/device/lib/runtime/log.h"
+#include "sw/device/lib/testing/test_framework/ujson_ottf.h"
+#include "sw/device/lib/ujson/ujson.h"
+#include "sw/device/tests/crypto/cryptotest/json/hash_commands.h"
+
+const int MaxMessageBytes = 17068;
+
+const int MaxDigestBytes = 256;
+const int MaxDigestWords = MaxDigestBytes / sizeof(uint32_t);
+
+const unsigned int kOtcryptoHashDigestWordsSha256 = 8;
```

Comment:
These constants are defined in [sha256.h](https://github.com/lowRISC/opentitan/blob/936d2a10e159e6096adec628ef69a67a82ec7ac5/sw/device/lib/crypto/impl/sha2/sha256.h#L52)

Reply:
Strangely, they only are for `SHA-256` and `SHA-512`. I still had to hard-code the others.